tools/lint

```diff
@@ -143,45 +150,47 @@ def run():
 
         lint_functions[name] = run_linter
 
-    external_linter('add_class', ['tools/find-add-class'])
-    external_linter('css', ['tools/check-css'], ['css'])
-    external_linter('eslint', ['node', 'node_modules/.bin/eslint', '--quiet'], ['js'])
-    external_linter('tslint', ['node', 'node_modules/.bin/tslint', '-c',
-                               'static/ts/tslint.json'], ['ts'])
-    external_linter('puppet', ['puppet', 'parser', 'validate'], ['pp'])
-    external_linter('templates', ['tools/check-templates'], ['handlebars', 'html'])
-    external_linter('urls', ['tools/check-urls'])
-    external_linter('swagger', ['node', 'tools/check-swagger'], ['yaml'])
+    if not args.backend:
+        external_linter('add_class', ['tools/find-add-class'])
+        external_linter('css', ['tools/check-css'], ['css'])
+        external_linter('eslint', ['node', 'node_modules/.bin/eslint', '--quiet'], ['js'])
+        external_linter('tslint', ['node', 'node_modules/.bin/tslint', '-c',
+                                   'static/ts/tslint.json'], ['ts'])
+        external_linter('templates', ['tools/check-templates'], ['handlebars', 'html'])
+        external_linter('urls', ['tools/check-urls'])
+    if not args.frontend:
+        external_linter('puppet', ['puppet', 'parser', 'validate'], ['pp'])
+        external_linter('swagger', ['node', 'tools/check-swagger'], ['yaml'])
```

Comment:
So, what I had in mind here, rather than explicitly declaring which linters to run, is to do this entirely by file type.  I've implemented my concept based on your work and merged it.